lib/python/picongpu/plugins/plot_mpl/energy_waterfall_visualizer.py

```diff
@@ -0,0 +1,173 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sophie Rudat, Sebastian Starke
+License: GPLv3+
+"""
+
+from picongpu.plugins.data import EnergyHistogramData
+from picongpu.plugins.plot_mpl.base_visualizer import Visualizer as\
+    BaseVisualizer, plt
+import numpy as np
+from matplotlib.colors import LogNorm
+
+
+class Visualizer(BaseVisualizer):
```

Comment:
Please adapt this class to the interface of the base class as well (most comments would be very similar to the ones I provided for the `emittance_evolution_visualizer.py`